src/test/java/com/google/sps/PlacesTest.java

```diff
@@ -88,9 +89,9 @@ public void getOnlyPlacesWithExpectedNumberOfRatingsAsHiddenGems() throws IOExce
 
   @Test 
   public void getRankedHiddenGemsBasedOnRatings() throws IOException {
-    try (LocalTestServerContext sc = new LocalTestServerContext(AllPlacesApiNearbySearchRequest)) {
-      Set<PlacesSearchResult> hiddenGems = Places.getAllHiddenGems(Places.fetchAllPlacesFromApi(sc.context, LOCATION));
-      List<PlacesSearchResult> rankedHiddenGems = Places.getRankedHiddenGems(hiddenGems);
+    try (LocalTestServerContext server = new LocalTestServerContext(AllPlacesApiNearbySearchRequest)) {
+      Set<HiddenGem> hiddenGems = Places.fetchHiddenGemsFromApi(server.context, Places.getAllHiddenPlaces(Places.fetchAllPlacesFromApi(server.context, LOCATION)));
+      List<HiddenGem> rankedHiddenGems = Places.getRankedHiddenGems(hiddenGems);
       for (int i = 0; i < rankedHiddenGems.size()-1; i++) {
         assertTrue(rankedHiddenGems.get(i).rating >= rankedHiddenGems.get(i + 1).rating);
       }
```

Comment:
Maybe add a TODO: Write test to check that we properly convert a PlacesSearchResult to a Hidden Gem object?

Reply:
done